src/Magnum/Platform/EmscriptenApplication.cpp

```diff
@@ -0,0 +1,474 @@
+    const size_t codeLength = std::strlen(code);
+    if(Utility::String::viewBeginsWith({code, codeLength}, "Digit")) {
+        return Key(code[5]);
+
+    /* Numpad keys */
+    } else if(Utility::String::viewBeginsWith({code, codeLength}, "Numpad")) {
+        std::string numKey(code + 6);
+        if(numKey == "Add") return Key::NumAdd;
+        if(numKey == "Decimal") return Key::NumDecimal;
+        if(numKey == "Divide") return Key::NumDivide;
+        if(numKey == "Enter") return Key::NumEnter;
+        if(numKey == "Equal") return Key::NumEqual;
+        if(numKey == "Multiply") return Key::NumMultiply;
+        if(numKey == "Subtract") return Key::NumSubtract;
+
+        /* Numpad0 - Numpad9 */
+        const Int num = numKey[0] - '0';
+        if(num >= 0 && num <= 9) {
+            return Key(num + Int(Key::NumZero));
+        }
+
+        return Key::Unknown;
+    }
+
+    const auto mapping = Containers::arrayView(Implementation::KEY_MAPPING,
+        Containers::arraySize(Implementation::KEY_MAPPING));
+    const Implementation::Entry* found =
+        std::lower_bound(mapping.begin(), mapping.end(), code, Implementation::EntryCompare{});
+    if(found != mapping.end() && std::strcmp(found->key, code) == 0) {
+        return found->value;
+    }
+
+    /* F1 - F12 */
+    if(code[0] == 'F') {
+        /* F1-F9 */
+        if(code[2] != '\0') {
+            const Int num = code[2] - '0';
+            return Key(Int(Key::F10) + num);
+        }
+        /* F10-F12 */
+        const Int num = code[1] - '1';
+        return Key(Int(Key::F1) + num);
+    }
+
+    return Key::Unknown;
+}
+
+void EmscriptenApplication::startTextInput() {
+    _flags |= Flag::TextInputActive;
+}
+
+void EmscriptenApplication::stopTextInput() {
+    _flags &= ~Flag::TextInputActive;
+}
+
+void EmscriptenApplication::setTextInputRect(const Range2Di&) {
+    // TODO: Place a hidden input field at given rect
+}
+
+void EmscriptenApplication::viewportEvent(ViewportEvent&) {}
+void EmscriptenApplication::keyPressEvent(KeyEvent&) {}
+void EmscriptenApplication::keyReleaseEvent(KeyEvent&) {}
+void EmscriptenApplication::mousePressEvent(MouseEvent&) {}
+void EmscriptenApplication::mouseReleaseEvent(MouseEvent&) {}
+void EmscriptenApplication::mouseMoveEvent(MouseMoveEvent&) {}
+void EmscriptenApplication::mouseScrollEvent(MouseScrollEvent&) {}
+void EmscriptenApplication::textInputEvent(TextInputEvent&) {}
+
+EmscriptenApplication::GLConfiguration::GLConfiguration():
+    _colorBufferSize{8, 8, 8, 0}, _depthBufferSize{24}, _stencilBufferSize{0} {}
+
+void EmscriptenApplication::mainLoopIteration() {
+    /* The resize event is not fired on window resize, so poll for the canvas
+       size here. But only if the window was requested to be resizable, to
+       avoid resizing the canvas when the user doesn't want that. Related
+       issue: https://github.com/kripken/emscripten/issues/1731 */
```

Comment:
This AFAIK holds for `emscripten_set_resize_callback()` as well, right? Maybe update the comment to make it clear it's not just a copypasted assumption from Sdl2App.

Reply:
Good question, I think it may actually fire even.

Reply:
Yes, it holds, see the [note in the emscripten documentation](https://emscripten.org/docs/api_reference/html5.h.html#c.emscripten_set_scroll_callback)

Reply:
Hmm. That led me to think we could attach the callback to the window instead of this ugly polling. Will try this locally.